include/directx/d3dx12.h

```diff
@@ -4037,6 +4038,277 @@ class CD3DX12_NODE_MASK_SUBOBJECT
     D3D12_NODE_MASK m_Desc;
 };
 
+//------------------------------------------------------------------------------------------------
+// Peter: Implementation for the new CheckFeatureSupport API
+
+#define FEATURE_SUPPORT_GET(RETTYPE,FEATURE,OPTION) \
+RETTYPE OPTION() const \
+{ \
+    return FEATURE.OPTION; \
+}
+
+#define FEATURE_SUPPORT_GET_NAME(RETTYPE,FEATURE,OPTION,NAME) \
+RETTYPE NAME() const \
+{\
+    return FEATURE.OPTION; \
+}
+
+#define FEATURE_SUPPORT_GET_NODE_INDEXED(RETTYPE,FEATURE,OPTION) \
+RETTYPE OPTION(UINT NodeIndex=0) const \
+{\
+    return FEATURE[NodeIndex].OPTION; \
+}
+
+#define INITIALIZE_MEMBER_DATA_CHECKED(FEATURE, MEMBER) \
+if (FAILED(result = m_pDevice->CheckFeatureSupport(FEATURE, &MEMBER, sizeof(MEMBER)))) { \
+    return result; \
+}
+
+class CD3DX12FeatureSupport
+{
+public:
+    CD3DX12FeatureSupport(ID3D12Device* pDevice)
+    : m_pDevice(pDevice)
+    {
+        m_hStatus = Init();
+    }
+
+    HRESULT Init()
+    {
+        // TODO: Think about hot to catch and report errors
+        HRESULT result;
+        
+        // Initialize static feature support data structures
+        INITIALIZE_MEMBER_DATA_CHECKED(D3D12_FEATURE_D3D12_OPTIONS, m_dOptions);
+        INITIALIZE_MEMBER_DATA_CHECKED(D3D12_FEATURE_GPU_VIRTUAL_ADDRESS_SUPPORT, m_dGPUVASupport);
+        m_dShaderModel.HighestShaderModel = D3D_HIGHEST_SHADER_MODEL; // TODO: What to do if highest model is inconsistent in the DLL and in the header?

```

Comment:
I ran into a problem where the d3d12.h header in the repo has more entries than the D3D12 runtime on my local machine. In this case the D3D_HIGHEST_SHADER_MODEL is not recognized and the initialization would fail. I can think of several ways to deal with this issue:
1. Try out every enum entry in reverse order until one version is supported. Store that version entry.
2. Do not initialize the data field during construction. Let the user provide the version they want in the getter API.

Reply:
Yeah, option 1 is the correct option here, and is one of the primary pieces of value that this helper can provide.

Reply:
Thanks for the advice! I guess that's how users' can feel comfortable using the new APIs. :)